cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -321,6 +321,20 @@ func stringList(s string) []string {
 	return spl
 }
 
+func checkInstanceConnectivity(instanceName string, client *proxy.Client) (error) {
+	// The connectivity test right now is just a basic dial to make sure it is reachable
+	// and nothing more. This will help rule out basic network connectivity problems, though
+	// such as firewalls and the like. 
+	conn, err := client.Dial(instanceName)
+	
+	if err != nil {
+		return fmt.Errorf("Performed connectivity test to %v and got an error: %v", instanceName, err)
+	}
+
+	conn.Close()
```

Comment:
This should be deferred (and called after L328) so it always happens.

Reply:
Will this work if `conn` was null in the case of an error?

Reply:
Go doesn't have `null`, it has `nil`. The main difference is that `nil` is a "zero value" and safe to use for things like this.

Reply:
Got it; I need to read me some more Go docs.